File: codingame/easy/onboarding/onboarding.pl
use strict;
use warnings;
#use diagnostics;
use 5.20.1;

select(STDOUT); $| = 1; # DO NOT REMOVE

# CodinGame planet is being attacked by slimy insectoid aliens.
# <---
# Hint:To protect the planet, you can implement the pseudo-code provided in the statement, below the player.


# game loop
while (1) {
    chomp(my $enemy_1 = <STDIN>); # name of enemy 1
    chomp(my $dist_1 = <STDIN>); # distance to enemy 1
    chomp(my $enemy_2 = <STDIN>); # name of enemy 2
    chomp(my $dist_2 = <STDIN>); # distance to enemy 2
    
    # Write an action using print
    # To debug: print STDERR "Debug messages...\n";

    # You have to output a correct ship name to shoot ("Buzz", enemy1, enemy2, ...)
    $_ = ($dist_1 lt $dist_2) ? $enemy_1 : $enemy_2;
    print "$_\n";
}

File: codingame/easy/onboarding/onboarding.scala
import math._
import scala.util._

/**
 * CodinGame planet is being attacked by slimy insectoid aliens.
 * <---
 * Hint:To protect the planet, you can implement the pseudo-code provided in the statement, below the player.
 **/
object Player extends App {

    // game loop
    while(true) {
        val enemy1 = readLine // name of enemy 1
        val dist1 = readInt // distance to enemy 1
        val enemy2 = readLine // name of enemy 2
        val dist2 = readInt // distance to enemy 2
        
        // Write an action using println
        // To debug: Console.err.println("Debug messages...")
        
        if(dist1 < dist2) {
            println(enemy1);
        }else{
            println(enemy2);
        }
        // You have to output a correct ship name to shoot ("Buzz", enemy1, enemy2, ...)
        //println("name of the enemy")
    }
}